### ISSUE TITLE
Scheduler Event Failure

### ISSUE THREAD

**Issue by MichaelMatsiy** [dnnsoftware/Dnn.Platform#3885]
I'm trying to export a site from DNN instance and getting an issue while the process.
The task stuck in 'in progress' status.

AbsoluteURL:
DefaultDataProvider:DotNetNuke.Data.SqlDataProvider, DotNetNuke
ExceptionGUID:6486216c-b78b-423d-8f83-5b7d6cdac4ca
AssemblyVersion:
PortalId:-1
UserId:-1
TabId:-1
RawUrl:
Referrer:
UserAgent:
ExceptionHash:zpkdYz96lbDRmQAvaNvvzpU37hA=
Message:Method not found: 'Void Dnn.ExportImport.Dto.Pages.ExportTab.set_Tags(System.String)'.
StackTrace:
   at Dnn.ExportImport.Components.Services.PagesExportService.SaveExportPage(TabInfo tab)
   at Dnn.ExportImport.Components.Services.PagesExportService.ProcessExportPages()
   at Dnn.ExportImport.Components.Services.PagesExportService.ExportData(ExportImportJob exportJob, ExportDto exportDto)
   at Dnn.ExportImport.Components.Engines.ExportImportEngine.Export(ExportImportJob exportJob, ExportImportResult result, ScheduleHistoryItem scheduleHistoryItem)
   at Dnn.ExportImport.Components.Scheduler.ExportImportScheduler.DoWork()
InnerMessage:
InnerStackTrace:
Source:DotNetNuke.SiteExportImport
FileName:
FileLineNumber:0
FileColumnNumber:0

07/01/2020 14:57:31
Scheduler Event Failure

THREAD ID 100 TYPE Dnn.ExportImport.Components.Scheduler.ExportImportSched 
THREAD ID:100

TYPE:Dnn.ExportImport.Components.Scheduler.ExportImportScheduler, DotNetNuke.SiteExportImport
EXCEPTION:Method not found: 'Void Dnn.ExportImport.Dto.Pages.ExportTab.set_Tags(System.String)'.
RESCHEDULED FOR:7/1/2020 3:57:28 PM
SOURCE:STARTED_FROM_SCHEDULE_CHANGE
ACTIVE THREADS:0
FREE THREADS:5
READER TIMEOUTS:0
WRITER TIMEOUTS:0
IN PROGRESS:0
IN QUEUE:6

![image](https://user-images.githubusercontent.com/14907457/86248133-2193d580-bbb6-11ea-9787-2acecd6189a8.png)

![image](https://user-images.githubusercontent.com/14907457/86248189-36706900-bbb6-11ea-80f7-28df655906a4.png)

* [+] 09.06.01 latest supported release

* [+] Chrome
* [+] Internet Explorer 11

**Comment by sleupold** [comment]
is this a new install or an upgraded website?
did you check /bin folder for older dotnetnuke...dll files, which haven't been properly deleted or replaced?

**Comment by MichaelMatsiy** [comment]
> is this a new install or an upgraded website?
> did you check /bin folder for older dotnetnuke...dll files, which haven't been properly deleted or replaced?

It's an upgrade.

Do I need to check the version of each 'DNN.*' DLL whether it 9.6.1 or not?

**Comment by sleupold** [comment]
usually you'll notice it from the "last updated" date.
did you unblock downloaded upgrade package (in file properties) prior to unzipping the files contained?

**Comment by MichaelMatsiy** [comment]
> usually you'll notice it from the "last updated" date.
> did you unblock downloaded upgrade package (in file properties) prior to unzipping the files contained?

No, I didn't unblock package. 
I've just unzipped it to the IIS Site folder and run upgrade. Is it not correct?

I've started removing of the old DNN DLLs (I've started from 8.* version) and got an issue of the missed DLLs.

**Comment by sleupold** [comment]
if you check file properties and there is a warning, that the file was obtained from the internet, you would have an option to unblock it (checkbox or button, depends on Windows version). You need to unblock, otherwise, not all files might get overwritten.

**Comment by MichaelMatsiy** [comment]
> if you check file properties and there is a warning, that the file was obtained from the internet, you would have an option to unblock it (checkbox or button, depends on Windows version). You need to unblock, otherwise, not all files might get overwritten.

Looks like I don't have problems with it.
Okay, I will try to remove old assemblies. If it helps I will close the ticket.

Thanks

**Comment by MichaelMatsiy** [comment]
@sleupold 
by the way, could it be the root cause of the problem that the import of the site is going the second day already? I choose 4 pages and full export, and it is still in the process

**Comment by sleupold** [comment]
there might be an issue with .dll files, that should have been deleted or replaced during the upgrade.
Which version did you upgrade from?

**Comment by MichaelMatsiy** [comment]
> there might be an issue with .dll files, that should have been deleted or replaced during the upgrade.
> Which version did you upgrade from?

9.1.1 to 9.4.1 a month ago and 9.4.1 to 9.6.1 a week ago

**Comment by valadas** [comment]
>by the way, could it be the root cause of the problem that the import of the site is going the second day already? I choose 4 pages and full export mode, and it is still in the process

The import/export is driven by the scheduler, if you go there you will see an item for import/export and I believe it is set by default to run once per day. You can either run it manually or reduce that time to shorter.

**Comment by MichaelMatsiy** [comment]
> > by the way, could it be the root cause of the problem that the import of the site is going the second day already? I choose 4 pages and full export mode, and it is still in the process
> 
> The import/export is driven by the scheduler, if you go there you will see an item for import/export and I believe it is set by default to run once per day. You can either run it manually or reduce that time to shorter.

Yes, I've tried, but it helps for a couple of hours. Then it returns to the stuck state

**Comment by MichaelMatsiy** [comment]
@sleupold How I can update Skin objects?
When I'm pushing on the new version in the menu to update one I'm getting an error like this 

![image](https://user-images.githubusercontent.com/14907457/86578736-f7129580-bf84-11ea-8763-0fa1632a64cd.png)

Is it my internal bug due to some absent settings or I should create a new ticket for this issue?

**Comment by MichaelMatsiy** [comment]
> there might be an issue with .dll files, that should have been deleted or replaced during the upgrade.
> Which version did you upgrade from?

Here are DLLs that haven't been updated.
Also, I haven't found any dependencies in the custom modules to them.

If I remove all of them, the application will crash.

![image](https://user-images.githubusercontent.com/14907457/86591748-d2c1b380-bf9a-11ea-93f9-6e72b2f489dc.png)

What can you suggest me in this case?

**Comment by MichaelMatsiy** [comment]
@sleupold 

I've downloaded source code and build all required solutions by myself.
It helped with import/export but currently I have this text in the Tab 

![image](https://user-images.githubusercontent.com/14907457/86594129-259d6a00-bf9f-11ea-9cdb-957b3c08e1f2.png)

I'm not sure that it is right as far as 9.6.1 was not an alpha version, wasn't it?

**Comment by sleupold** [comment]
For DNN 9.6.x, you shouldn't need any of the 9.4.x and 9.0.x and 8.0.x dlls.
What is the error message, if you remove it?
All skin objects packed with DNN are updated automatically.
I wouldn't try to build DNN yourself - this won't help and cause more issues.

**Comment by MichaelMatsiy** [comment]
@sleupold 
These DLLs that you can see on the screen were not updated with the 9.6.1 package. 
They exist in the 9.4.1 package but not in the 9.6.1. 
As if I can see them in the source code in the project I can make an assumption that these DLL's are not deprecated, just missed.

When I'd removed a couple of them I've got issues like 500 or missed references/dependencies in the others DNN DLLs.

Then I've downloaded 9.6.1 source code and built all the required DLLs, now everything is fine and export was complete in a 5 seconds after scheduler restarting. As I see, it means that these DLL's are important, but after adding of them I've got this message in the tab about Alpha version

**Comment by sleupold** [comment]
@MichaelMatsiy 
there should be never versions of most dll files contained in the upgrade package - most packed in install zip files inside /install/module folder.

**Comment by MichaelMatsiy** [comment]
@sleupold 
Yes, I have.

I've tried to install them and now it is fine. 

Thanks

**Comment by MichaelMatsiy** [comment]
This bug can be closed as Not A Bug.

@sleupold thanks for your help

**Comment by sleupold** [comment]
@MichaelMatsiy 
I am glad, you were able to solve it.
Have fun using DNN :)
